python_modules/libraries/dagster-aws/dagster_aws/emr/cloud_watch_logs_follower.py

```diff
@@ -0,0 +1,90 @@
+from collections import defaultdict
+from typing import DefaultDict
+
+import botocore
+
+
+class CloudwatchLogsFollower:
+    """Tail CloudWatch Logs from log streams.
+
```

Comment:
nit: we should match docstring format from the rest of the codebase (single backticks, empty line at the top of the docstring, no `----` lines)

```py
"""
Tail Cloudwatch Logs from log streams.

CloudWatch Logs has no API to directly stream new log
events from log streams. Instead, we have to regularly
fetch new logs using an appropriate `startTime` argument.

When `log_stream_name_prefix` matches multiple log streams,
it is possible that some log events will be missed.
...
"""
```